libs/vkd3d/vkd3d_private.h

```diff
@@ -618,8 +623,32 @@ struct d3d12_descriptor_heap
 
 HRESULT d3d12_descriptor_heap_create(struct d3d12_device *device,
         const D3D12_DESCRIPTOR_HEAP_DESC *desc, struct d3d12_descriptor_heap **descriptor_heap) DECLSPEC_HIDDEN;
+void d3d12_descriptor_heap_cleanup(struct d3d12_descriptor_heap *descriptor_heap) DECLSPEC_HIDDEN;
 struct d3d12_descriptor_heap *unsafe_impl_from_ID3D12DescriptorHeap(ID3D12DescriptorHeap *iface) DECLSPEC_HIDDEN;
 
+static inline unsigned int d3d12_descriptor_heap_sampler_set_index()
+{
+    return 0;
+}
+
+static inline unsigned int d3d12_descriptor_heap_cbv_set_index()
+{
+    return 0;
+}
+
+static inline unsigned int d3d12_descriptor_heap_srv_set_index(bool is_buffer)
+{
+    return 1 + (is_buffer ? 0 : 1);
+}
+
+static inline unsigned int d3d12_descriptor_heap_uav_set_index(bool is_buffer)
+{
+    return 3 + (is_buffer ? 0 : 1);
```

Comment:
I suppose there is no argument for counters since we don't bother with bindless UAV counters yet?

Reply:
Kind of, in practice (assuming they get a separate set, or are incorporated into an existing set) we'd probably define a new function for UAV counters instead.